Lib/fontbakery/profiles/gdef.py

```diff
@@ -0,0 +1,116 @@
+from fontbakery.callable import check
+from fontbakery.checkrunner import FAIL, PASS, WARN
+from fontbakery.message import Message
+from fontbakery.utils import pretty_print_list
+
+# used to inform get_module_profile whether and how to create a profile
+from fontbakery.fonts_profile import profile_factory # NOQA pylint: disable=unused-import
+
+from fontTools import unicodedata
+
+@ check(
+  id = 'com.google.fonts/check/gdef_spacing_marks',
+  rationale ="""
+  Glyphs in the GDEF mark glyph class should be non-spacing.
+  Spacing glyphs in the GDEF mark glyph class may have incorrect
+  anchor positioning that was only intended for building composite glyphs
+  during design.
+  """
+)
+def com_google_fonts_check_gdef_spacing_marks(ttFont):
+  """Check mark characters are in GDEF mark glyph class)"""
+  if "GDEF" in ttFont:
+    spacing_glyphs = {name for (name, value) in ttFont["hmtx"].metrics.items()
+                      if value}
+    class_defs = ttFont["GDEF"].table.GlyphClassDef.classDefs.items()
+    mark_class_glyphs = set([name for (name, value) in class_defs if value == 3])
+    spacing_glyphs_in_mark_glyph_class = spacing_glyphs & mark_class_glyphs
+    if spacing_glyphs_in_mark_glyph_class :
+      formatted_list = "\t " +\
+        pretty_print_list(sorted(spacing_glyphs_in_mark_glyph_class),
+                          shorten=10,
+                          sep=", ")
+      yield WARN,\
+        Message('spacing glyphs in mark glyph class',
+        f"The following spacing glyphs may be in the GDEF mark glyph"
+        f"class by mistake:\n"
+        f"{formatted_list}")
+  else:
+    yield PASS, 'Font does not declare an optional "GDEF" table.'
+
+
+@check(
+  id = 'com.google.fonts/check/gdef_mark_chars',
+  rationale = """
+  Mark characters should be in the GDEF mark glyph class.
+  """,
+)
+def com_google_fonts_check_gdef_mark_chars(ttFont):
+  """Check mark characters are in GDEF mark glyph class"""
+
+  if "GDEF" in ttFont:
+    cmap = ttFont.getBestCmap()
+    mark_char_glyphs = set()
+    for c in cmap:
+      if unicodedata.category(chr(c)).startswith("M"):
+        mark_char_glyphs.add(cmap[c])
+    class_defs = ttFont["GDEF"].table.GlyphClassDef.classDefs.items()
+    mark_class_glyphs = set([name for (name, value) in class_defs if value == 3])
```

Comment:
Can we use a constant with a name that explains its meaning here instead of simply the number 3? If we still do not have a constant for that, please add one to `fontbakery.constants`. We prefer to avoid magic numbers, as the meaning of the source code routines should be as easy as possible to grasp at a quick glance.